src/Dfe.PlanTech.Web/TagHelpers/FooterLinkTagHelper.cs

```diff
@@ -53,14 +46,61 @@ private static void AppendCloseTag(StringBuilder stringBuilder)
 
     private void AppendOpenTag(StringBuilder stringBuilder)
     {
-        stringBuilder.Append("<a class=").Append(FOOTER_CLASS).Append(" href=\"");
-        stringBuilder.Append(Link!.Href);
+        stringBuilder.Append("""<a class="govuk-footer__link" """).Append(" href=\"");
+        stringBuilder.Append(GetHref());
         stringBuilder.Append('"');
 
-        if (Link.OpenInNewTab)
+        if (Link!.OpenInNewTab)
         {
             stringBuilder.Append(" target=\"_blank\"");
         }
+
         stringBuilder.Append('>');
     }
+
+    private string GetHref()
+    {
+        if (Link!.ContentToLinkTo == null && string.IsNullOrEmpty(Link.Href))
+        {
+            logger.LogError("No href or content to link to for {LinkType}", nameof(NavigationLink));
+            return string.Empty;
+        }
+
+        return GetUrlForContent() ?? Link.Href ?? string.Empty;
+    }
+
+    private string? GetUrlForContent()
+    {
+        if (Link?.ContentToLinkTo == null)
+        {
+            return null;
+        }
+
+        if (Link.ContentToLinkTo is not IHasSlug hasSlug)
+        {
+            logger.LogError("Invalid content type received for Link. Expected {Interface} but type is {Concrete}", typeof(IHasSlug), Link.ContentToLinkTo.GetType());
+            return null;
+        }
+
+        var firstCharacterIsSlash = hasSlug.Slug[0] == '/';
+
+        var slug = firstCharacterIsSlash ? hasSlug.Slug.AsSpan(1) : hasSlug.Slug.AsSpan();
+
+        return Link.ContentToLinkTo switch
+        {
+            IPage => $"/{slug}",
+            IContentSupportPage => $"/content/{slug}",
```

Comment:
This isn't linked to the content and support HelloWorld page ive just added in contentful, somethings going wrong

Reply:
Fixed I think